library/tiqr/tests/Tiqr_UserSecretStorage_PdoTest.php

```diff
@@ -80,6 +80,24 @@ public function test_it_can_store_and_retrieve_an_user_secret()
         $this->assertEquals('my-secret', $secret);
     }
 
+    public function test_error_reading_and_writing_user_secret()
+    {
+        $store = Tiqr_UserSecretStorage::getSecretStorage(
+            'pdo',
+            $this->logger,
+            array(
+                'dsn' => $this->dsn,
+                'username' => 'root',
+                'password' => 'secret',
+                'table' => 'tiqrusersecret_xxx'
+            )
+        );
+        $this->expectException(ReadWriteException::class);
+        $store->setSecret('user-id-1', 'my-secret');
+        $this->expectException(ReadWriteException::class);
+        $store->getSecret('user-id-1');
```

Comment:
I'm currently not set up to to run this test without hassle. I'm curious. Is the ReadWriteException message relevant to just log to the application log?

And I'm usually in the habit to also check if the expected exception message matches the actual message being thrown. That way: 

1. The reader of the test is informed about the error situation we are testing (assuming the exception message is descriptive)
2. The test becomes more specific. In your case the `ReadWriteException` you are catching might have different messages for your two different test scenarios. But this is not verified here.

I'm fine with leaving it like this. But it might be something to consider.

Reply:
All I want to test here is that an exception is thrown by PDO when the query fails. The ReadWriteException is converted from the PDO exception such that it will contain the same message. The actual message will depend on the PDO driver used and the actual error encountered. So in real scenario's that message should be informative.